src/wpool_pool.erl

```diff
@@ -220,32 +220,36 @@ stats(Wpool, Sup) ->
   {Total, WorkerStats} =
     lists:foldl(
       fun(N, {T, L}) ->
-        Worker = erlang:whereis(worker_name(Sup, N)),
-        [{message_queue_len, MQL} = MQLT,
-         Memory, Function, Location, {dictionary, Dictionary}] =
-          erlang:process_info(
-            Worker,
-            [ message_queue_len
-            , memory
-            , current_function
-            , current_location
-            , dictionary
-            ]),
-        Time =
-          calendar:datetime_to_gregorian_seconds(calendar:universal_time()),
-        WS =
-          case {Function, proplists:get_value(wpool_task, Dictionary)} of
-            {{current_function, {gen_server, loop, 6}}, undefined} ->
-              [MQLT, Memory];
-            {{current_function, {erlang, hibernate, _}}, undefined} ->
-              [MQLT, Memory];
-            {_, undefined} ->
-              [MQLT, Memory, Function, Location];
-            {_, {_TaskId, Started, Task}} ->
-              [MQLT, Memory, Function, Location,
-               {task, Task}, {runtime, Time - Started}]
-          end,
-        {T + MQL, [{N, WS} | L]}
+        case erlang:whereis(worker_name(Sup, N)) of
+          undefined ->
+            {T, L};
+          Worker ->
+            [{message_queue_len, MQL} = MQLT,
+             Memory, Function, Location, {dictionary, Dictionary}] =
+              erlang:process_info(
+                Worker,
+                [ message_queue_len
+                , memory
+                , current_function
+                , current_location
+                , dictionary
+                ]),
+            Time =
+              calendar:datetime_to_gregorian_seconds(calendar:universal_time()),
+            WS =
+              case {Function, proplists:get_value(wpool_task, Dictionary)} of
```

Comment:
The expression on line 240 and column 15 is nested beyond the maximum level of 3.